proxy/http3/Http3Transaction.cc

```diff
@@ -342,30 +429,16 @@ Http3Transaction::~Http3Transaction()
 }
 
 int
-Http3Transaction::state_stream_open(int event, void *edata)
+Http3Transaction::state_stream_open(int event, Event *edata)
 {
   // TODO: should check recursive call?
-  if (this->_thread != this_ethread()) {
```

Comment:
Note for reviewers: This block was copied from H2 but thread mismatch should not happen. If it happens, we should fix the code that schedules events on a wrong thread.